### PR TITLE
OmegaConf.update merge flag is now True by default

### LINKED ISSUE #667: Change the default value of OmegaConf.update merge flag to True
The merge flag was added in OmegaConf 2.0.2 with the intention of making it default to True.
to avoid a breaking change, a warning was issued for people that used update and did not specify a value for the merge flag.

It's time to clean this up.

### PR DESCRIPTION
**Pull request by omry**
closes #667

See issue for some context.

### CODE REVIEW

**Review by Jasha10 — APPROVED** [review]

File: docs/source/usage.rst [hunk 1]
```diff
@@ -898,25 +898,26 @@ OmegaConf.update
...
     >>> cfg = OmegaConf.create({"foo" : {"bar": 10}})
-    >>> # Merge flag has no effect because the value is a primitive
-    >>> OmegaConf.update(cfg, "foo.bar", 20, merge=True)
+    >>> OmegaConf.update(cfg, "foo.bar", 20)
     >>> assert cfg.foo.bar == 20
     >>> # Set dictionary value (using dot notation)
     >>> OmegaConf.update(cfg, "foo.bar", {"zonk" : 30}, merge=False)
     >>> assert cfg.foo.bar == {"zonk" : 30}
-    >>> # Merge dictionary value (using bracket notation)
+    >>> # Merge dictionary value (using bracket notation)\
```
Comment (omry) [inline docs/source/usage.rst:913]:
```suggestion
    >>> # Merge dictionary value (using bracket notation)
```

File: news/667.api_change [hunk 2]
```diff
@@ -0,0 +1 @@
+OmegaConf.update() is now merging dict/list values into the destination node by default. call with merge=False to replace instead.
```
Comment (odelalleau) [inline news/667.api_change:1]: Minor case issue:
```suggestion
OmegaConf.update() is now merging dict/list values into the destination node by default. Call with merge=False to replace instead.
```

File: tests/test_update.py [hunk 3]
```diff
@@ -180,22 +179,15 @@ def test_update_list_index_error() -> None:
...
-    msg = dedent(
-        """\
-            update() merge flag is is not specified, defaulting to False.
-            For more details, see https://github.com/omry/omegaconf/issues/367"""
-    )
-
-    with warns(UserWarning, match=re.escape(msg)):
-        OmegaConf.update(cfg, "a", {"c": 20})  # default to set, and issue a warning.
-        assert cfg == {"a": {"c": 20}}
+    OmegaConf.update(cfg, "a", {"c": 20})  # default to set, and issue a warning.
```
Comment (odelalleau) [inline tests/test_update.py:189]:
```suggestion
    OmegaConf.update(cfg, "a", {"c": 20})
```

**Review by odelalleau — CHANGES_REQUESTED** [review]
Go ahead and merge after fixing these two minor things

**Pull request merged**